test/algconvergence/ode_quadruple_precision_tests.jl

```diff
@@ -0,0 +1,59 @@
+using OrdinaryDiffEq, DiffEqDevTools, Test, Random
+import ODEProblemLibrary: prob_ode_linear, prob_ode_2Dlinear, prob_ode_bigfloat2Dlinear
+
+Random.seed!(100)
+
+dts = 1 .// 2 .^ (8:-1:4)
+testTol = 0.25
+
+f = (u, p, t) -> cos(t)
+prob_ode_sin = ODEProblem(ODEFunction(f; analytic = (u0, p, t) -> sin(t)), 0.0, (0.0, 1.0))
+
+f = (du, u, p, t) -> du[1] = cos(t)
+prob_ode_sin_inplace = ODEProblem(ODEFunction(f; analytic = (u0, p, t) -> [sin(t)]), [0.0],
+    (0.0, 1.0))
+
+f = (u, p, t) -> sin(u)
+prob_ode_nonlinear = ODEProblem(
+    ODEFunction(f;
+        analytic = (u0, p, t) -> 2 * acot(exp(-t) *
+                                          cot(0.5))), 1.0,
+    (0.0, 0.5))
+
+f = (du, u, p, t) -> du[1] = sin(u[1])
+prob_ode_nonlinear_inplace = ODEProblem(
+    ODEFunction(f;
+        analytic = (u0, p, t) -> [
+            2 * acot(exp(-t) * cot(0.5))
+        ]),
+    [1.0], (0.0, 0.5))
+
+test_problems_only_time = [prob_ode_sin, prob_ode_sin_inplace]
+test_problems_linear = [prob_ode_linear, prob_ode_2Dlinear, prob_ode_bigfloat2Dlinear]
+test_problems_nonlinear = [prob_ode_nonlinear, prob_ode_nonlinear_inplace]
+
+f_ssp = (u, p, t) -> begin
+    sin(10t) * u * (1 - u)
+end
+test_problem_ssp = ODEProblem(f_ssp, 0.1, (0.0, 8.0))
+test_problem_ssp_long = ODEProblem(f_ssp, 0.1, (0.0, 1.e3))
+
+f_ssp_inplace = (du, u, p, t) -> begin
+    @. du = sin(10t) * u * (1 - u)
+end
+test_problem_ssp_inplace = ODEProblem(f_ssp_inplace, rand(3, 3), (0.0, 8.0))
+
+println("QPRK98")
+alg = QPRK98()
+for prob in test_problems_only_time
+    sim = test_convergence(dts, prob, alg)
+    @test sim.𝒪est[:final]≈OrdinaryDiffEq.alg_order(alg) atol=testTol
+end
+for prob in test_problems_linear
```

Comment:
This is missing a regression test on the adaptivity. Solve with adaptivity and check the solution is sufficiently small. Check Vern9 as an example

Reply:
Ok I will do that. But is there Vern9 example in test, I don't find it ?

Reply:
https://github.com/SciML/OrdinaryDiffEq.jl/blob/master/test/regression/ode_unrolled_comparison_tests.jl

Reply:
Ok Thank you

Reply:
So I have done some tests with adaptivity and it seems to work fine, but the (automatic) interpolation doesn't, but may be that is normal and another unrelated problem : 
```julia 
f = (u, p, t) -> cos(t)
prob_ode_sin = ODEProblem(
                    ODEFunction(f; analytic = (u0, p, t) -> sin(t)), 
                    Float128(0.0), 
                    (Float128(0.0), Float128(4.0)))

alg = QPRK98()
sol = solve(prob_ode_sin, alg, dt = 1 / 2^10, adaptive = true)

plot(sol)
plot!(LinRange(0,4,1000), sin.(LinRange(0,4,1000)), label = "exact solution")
scatter!(sol.t, sin.(sol.t), label = "Real points computed", mc = :yellow)
```
![image](https://github.com/SciML/OrdinaryDiffEq.jl/assets/42682941/2e023685-37f8-4609-9926-d60aa968edf8)

Reply:
Another remark concerning tests is that the order of convergence is each time higer (from 10 to 11.5 instead of 9) for problems I chose, I don't know if this is normal

Reply:
> Are you updating fsal first and last to be the derivative estimates at 0 and dt?

I don't implement fsal first and last, I will try to add this

Reply:
> Another remark concerning tests is that the order of convergence is each time higer (from 10 to 11.5 instead of 9) for problems I chose, I don't know if this is normal

Yeah that's just super convergence. It's typical that extremely simple ODEs can hit higher orders since the next order conditions can just be zero "accidentally" if certain properties like df/du = 0.

Reply:
> Are you updating fsal first and last to be the derivative estimates at 0 and dt?

Seems better now :

![image](https://github.com/SciML/OrdinaryDiffEq.jl/assets/42682941/cc4d4f31-3794-4ec9-8b3c-8836487712ca)

> Yeah that's just super convergence. It's typical that extremely simple ODEs can hit higher orders since the next order conditions can just be zero "accidentally" if certain properties like df/du = 0.

Ok yes I see, thanks for the precision